enforcer/lookup/lookup.go

```diff
@@ -140,15 +139,23 @@ func (m *PolicyDB) AddPolicy(selector policy.TagSelector) (policyID int) {
 
 }
 
-func (m *PolicyDB) keyValueFromString(tag string) (key, value string) {
-
-	parts := strings.SplitN(tag, "=", 2)
+// Custom implementation for splitting strings. Gives significant performance
+// improvement. Do not allocate new strings
+func (m *PolicyDB) tagSplit(str string, k *string, v *string) error {
+	n := len(str)
+	if n == 0 {
+		return fmt.Errorf("Null string")
+	}
 
-	if len(parts) != 2 {
-		return "", ""
+	for i := 0; i < n; i++ {
+		if str[i] == '=' {
+			*k = str[:i]
+			*v = str[i+1:]
+			return nil
+		}
 	}
 
-	return parts[0], parts[1]
+	return fmt.Errorf("No key/value pair")
```

Comment:
Lowercase, and give the actual value of the tag we tried to split. This will help